commcare_connect/form_receiver/processor.py

```diff
@@ -133,7 +138,13 @@ def process_deliver_form(user, xform: XForm, app: CommCareApp, opportunity: Oppo
 
 def process_deliver_unit(user, xform: XForm, app: CommCareApp, opportunity: Opportunity, deliver_unit_block: dict):
     deliver_unit = get_or_create_deliver_unit(app, deliver_unit_block)
-    UserVisit.objects.create(
+    counts = UserVisit.objects.filter(opportunity=opportunity, user=user).aggregate(
+        daily=Count("pk", filter=Q(visit_date__date=datetime.date.today())), total=Sum("pk", default=0)
```

Comment:
Apologies for missing this the first time, I think this needs to use the date of actual delivery (form date), rather than date we receive it, which could happen at a delay due to connectivity issues. I am not sure we save that now so we may need to update the model as well, but in short if you do one visit on three consecutive days, but submit all those forms at once on the third day, all should count since they were done on different days.

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/146/commits/92a105a5b74eaaa92e080bc05d358c8432489c5f

I have added the check to count visits on the actual visit_date of the form that we are currently processing. We save the form visit_date on the UserVisit model already.